### PR TITLE
1 - add mongo_url parameter as argparse option to allow for connection to a mongo cluster with replicaset

### PR DESCRIPTION
**Pull request by dave00galloway**

### CODE REVIEW

File: benchrun.py [hunk 1]
```diff
@@ -145,29 +155,32 @@ def main():
...
         print("         Benchrun will continue without authentication.")
 
-    if args.includeFilter == [] :
+    if not args.includeFilter:
         args.includeFilter = '%'
-    elif len(args.includeFilter) == 1 :
+    elif len(args.includeFilter) == 1:
         args.includeFilter = args.includeFilter[0]
-        if args.includeFilter == ['%'] :
+        if args.includeFilter == ['%']:
```
Comment (henrikingo) [inline benchrun.py]: Not yours, but how could this if be true, given the previous 2 lines?

File: benchrun.py [hunk 2]
```diff
@@ -145,29 +155,32 @@ def main():
 
     auth = []
     using_auth = False
-    if isinstance(args.username, basestring) and isinstance(args.password, basestring):
+    # noinspection PyTypeChecker
```
Comment (henrikingo) [inline benchrun.py]: Sorry but please remove this comment, as PyTypeChecker is not a tool actually used in this repo. Same for other instances of this comment.

Comment (dave00galloway) [reply benchrun.py]: fair enough.

File: benchrun.py (hunk 1 above)
Comment (dave00galloway) [reply benchrun.py]: not sure exactly but maybe this is testing for someone having passed in the string '%' as the includeFilter. Let's see what happens when your internal tests run!

**Comment by dave00galloway** [comment]
contributor agreement signed, and pycharm specific comments removed as requested :)

**Comment by dave00galloway** [comment]
@henrikingo - is this ok or are further changes required?

**Comment by henrikingo** [comment]
Hi

I think this is good from your side. I have created a jira ticket for our team to run some tests on this and if they pass it can be merged. I won't click "approve" until then, but there's nothing more for you to do.

Our iteration cycle is 2 weeks, so it will take a multiple of 2 weeks before you see any activity from our side.

**Comment by henrikingo** [comment]
Hi Dave

I'm terribly sorry, but this minor task got buried in some backlog a year ago. I have now picked it up.

The good news is that the tests seem to run and pass.

I can't find a Dave Galloway in the list of signed contributor agreements. Could you please try again? You can email me at henrik.ingo@mongodb.com if there's any issue with that.

Again, sorry about the delay here.

**Comment by vivian-ge** [comment]
Closing due to inactivity - please open a new PR if you’d like to contribute. Thank you!

**Pull request closed without merging**